### PR TITLE
Inference CLI validation for Scoring FE

### PR DESCRIPTION
**Pull request by liakaz**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by jonathan-innis** [comment]
Test cases are failing with `ERROR: Provide ssl certificate and key. Otherwise explicitly allow insecure connection by specifying '--configuration-settings allowInsecureConnections=true'` You can update test cases by creating a PR against the [K8sPartnerExtensionTest Repo](https://msazure.visualstudio.com/DefaultCollection/One/_git/compute-HybridMgmt-K8sPartnerExtensionTest)

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py [hunk 1]
```diff
@@ -171,12 +171,44 @@ def __validate_config(self, configuration_settings, configuration_protected_sett
...
+        allowInsecureConnections = _get_value_from_config_protected_config(
+            'allowInsecureConnections', configuration_settings, configuration_protected_settings)
+        allowInsecureConnections = str(allowInsecureConnections).lower() == 'true'
+        if (not feSslCert or not sslKey) and not allowInsecureConnections:
+            raise InvalidArgumentValueError(
+                "Provide ssl certificate and key. "
+                "Otherwise explicitly allow insecure connection by specifying "
+                "'--configuration-settings allowInsecureConnections=true'")
+
+        feIsNodeport = _get_value_from_config_protected_config(
```
Comment (jonathan-innis) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:203]: Style check is failing here with unused variable

**Review by jonathan-innis — CHANGES_REQUESTED** [review]

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py (hunk 1 above)
Comment (liakaz) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py]: fixed. I don't actually think there is any validation on this that we need to do. Pulled it up by accident.

**Review by jonathan-innis — APPROVED** [review]
🚀

**Review by CindyXing — APPROVED** [review]

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py [hunk 2]
```diff
@@ -171,12 +171,42 @@ def __validate_config(self, configuration_settings, configuration_protected_sett
...
         configuration_settings[self.ENABLE_TRAINING] = configuration_settings.get(self.ENABLE_TRAINING, enable_training)
         configuration_settings[self.ENABLE_INFERENCE] = configuration_settings.get(
             self.ENABLE_INFERENCE, enable_inference)
         configuration_protected_settings.pop(self.ENABLE_TRAINING, None)
         configuration_protected_settings.pop(self.ENABLE_INFERENCE, None)
 
+    def __validate_scoring_fe_settings(self, configuration_settings, configuration_protected_settings):
+        clusterPurpose = _get_value_from_config_protected_config(
+            'clusterPurpose', configuration_settings, configuration_protected_settings)
+        if clusterPurpose and clusterPurpose not in ["DevTest", "FastProd"]:
```
Comment (mjaow) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:185]: there is alway another value "test"

Comment (liakaz) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:185]: fixed in helm chart. Should be either DevTest or FastProd

**Pull request merged**